src/Sulu/Bundle/MediaBundle/DependencyInjection/StorageCompilerPass.php

```diff
@@ -0,0 +1,221 @@
+<?php
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MediaBundle\DependencyInjection;
+
+use Sulu\Bundle\MediaBundle\Media\Exception\StorageAdapterNotFoundException;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\DefinitionDecorator;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Compiler to create flysystem storage and add storage to the manager.
+ */
+class StorageCompilerPass implements CompilerPassInterface
+{
+    /**
+     * @var array
+     */
+    protected $adapters = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('sulu_media.storage_manager')) {
+            return;
+        }
+
+        // Flysystem
+        $this->addStorageAdapter(
+            $container,
+            'flysystem',
+            'League\Flysystem\MountManager',
+            'oneup_flysystem.mount_manager',
+            'Sulu\Bundle\MediaBundle\Media\Storage\Resolver\FlysystemResolver',
+            [
+                's3' => [
+                    'class' => 'Sulu\Bundle\MediaBundle\Media\Storage\Resolver\Flysystem\AwsS3Resolver',
+                    'tag' => 'League\Flysystem\AwsS3v3\AwsS3Adapter',
+                ],
+                'dropbox' => [
+                    'class' => 'Sulu\Bundle\MediaBundle\Media\Storage\Resolver\Flysystem\DropBoxResolver',
+                    'tag' => 'League\Flysystem\AwsS3v3\DropboxAdapter',
+                ],
```

Comment:
first I had only one resolver class but did talk then with @chirimoya that if somebody add a new flysystem adapter he maybe need to create also a new resolver so I created foreach adapter an own resolver. 
The resolver are tagged with name=`sulu_media.storage.resolver.flysystem` alias=`class` to class is needed to now for which adapter the resolver is.